The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19430984273/job/55589748708) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/55589748708) [(plain)](https://github.com/rust-lang/rust/commit/c51597470bd62768497068cb38a7dac78b9a2fbe/checks/55589748708/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs:13:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: unwrap_failed
               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/issues/iter-max-no-unwrap-failed-129583/iter-max-no-unwrap-failed-129583.ll:44:213: note: found here
 br i1 %_25.i.2, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb10.i
                                                                                                                                                                                                                    ^~~~~~~~~~~~~

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/issues/iter-max-no-unwrap-failed-129583/iter-max-no-unwrap-failed-129583.ll
Check file: /checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
        1: ; ModuleID = 'iter_max_no_unwrap_failed_129583.5c466b832002e549-cgu.0' 
        2: source_filename = "iter_max_no_unwrap_failed_129583.5c466b832002e549-cgu.0" 
        3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-i128:128-n32:64-S128-ni:1:10:20" 
        4: target triple = "wasm32-unknown-wasip1" 
        5:  
        6: @alloc_be0d87cab5c3b8ce747aaa7e9520f67f = private unnamed_addr constant [72 x i8] c"/checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs\00", align 1 
        7: @alloc_ec844ed9573175be2fa8ca9555f08e2e = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_be0d87cab5c3b8ce747aaa7e9520f67f, [12 x i8] c"G\00\00\00\1D\00\00\00\15\00\00\00" }>, align 4 
        8:  
        9: ; Function Attrs: nofree norecurse nosync nounwind memory(argmem: read, inaccessiblemem: readwrite) 
       10: define dso_local noundef i8 @infallible_max_not_unrolled(ptr noalias noundef readonly align 1 captures(none) dereferenceable(1024) %x) unnamed_addr #0 { 
       11: start: 
       12:  %_23 = getelementptr inbounds nuw i8, ptr %x, i32 1 
       13:  tail call void @llvm.experimental.noalias.scope.decl(metadata !1) 
       14:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x, align 1, !alias.scope !4, !noalias !9 
       15:  %invariant.gep = getelementptr inbounds i8, ptr %_23, i32 1 
       16:  %invariant.gep1 = getelementptr inbounds i8, ptr %_23, i32 2 
       17:  br label %bb10.i 
       18:  
       19: bb10.i: ; preds = %bb10.i, %start 
       20:  %_3.val.i.i.i.i.i.i = phi i8 [ %_3.val.i.i.i.i.i.pre.i, %start ], [ %2, %bb10.i ] 
       21:  %i.sroa.0.0.i = phi i32 [ 0, %start ], [ %_24.i.2, %bb10.i ] 
       22:  %init.sroa.0.0.i = phi ptr [ %x, %start ], [ %..i.i.i.2, %bb10.i ] 
       23:  %_31.i = getelementptr inbounds nuw i8, ptr %_23, i32 %i.sroa.0.0.i 
       24:  tail call void @llvm.experimental.noalias.scope.decl(metadata !12) 
       25:  tail call void @llvm.experimental.noalias.scope.decl(metadata !13) 
       26:  tail call void @llvm.experimental.noalias.scope.decl(metadata !14) 
       27:  tail call void @llvm.experimental.noalias.scope.decl(metadata !15) 
       28:  %_4.val.i.i.i.i.i.i = load i8, ptr %_31.i, align 1, !alias.scope !9, !noalias !4, !noundef !16 
       29:  %_4.i.i.i = icmp ugt i8 %_3.val.i.i.i.i.i.i, %_4.val.i.i.i.i.i.i 
       30:  %0 = tail call i8 @llvm.umax.i8(i8 %_3.val.i.i.i.i.i.i, i8 %_4.val.i.i.i.i.i.i) 
       31:  %..i.i.i = select i1 %_4.i.i.i, ptr %init.sroa.0.0.i, ptr %_31.i 
       32:  %gep = getelementptr inbounds i8, ptr %invariant.gep, i32 %i.sroa.0.0.i 
       33:  %_4.val.i.i.i.i.i.i.1 = load i8, ptr %gep, align 1, !alias.scope !17, !noalias !20, !noundef !16 
       34:  %_4.i.i.i.1 = icmp ugt i8 %0, %_4.val.i.i.i.i.i.i.1 
       35:  %1 = tail call i8 @llvm.umax.i8(i8 %0, i8 %_4.val.i.i.i.i.i.i.1) 
       36:  %..i.i.i.1 = select i1 %_4.i.i.i.1, ptr %..i.i.i, ptr %gep 
       37:  %gep2 = getelementptr inbounds i8, ptr %invariant.gep1, i32 %i.sroa.0.0.i 
       38:  %_4.val.i.i.i.i.i.i.2 = load i8, ptr %gep2, align 1, !alias.scope !23, !noalias !26, !noundef !16 
       39:  %_4.i.i.i.2 = icmp ugt i8 %1, %_4.val.i.i.i.i.i.i.2 
       40:  %2 = tail call i8 @llvm.umax.i8(i8 %1, i8 %_4.val.i.i.i.i.i.i.2) 
       41:  %..i.i.i.2 = select i1 %_4.i.i.i.2, ptr %..i.i.i.1, ptr %gep2 
       42:  %_24.i.2 = add nuw nsw i32 %i.sroa.0.0.i, 3 
       43:  %_25.i.2 = icmp eq i32 %_24.i.2, 1023 
       44:  br i1 %_25.i.2, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb10.i 
not:13                                                                                                                                                                                                                         !~~~~~~~~~~~~                             error: no match expected
       45:  
       46: _RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit: ; preds = %bb10.i 
       47:  %_0 = load i8, ptr %..i.i.i.2, align 1, !noundef !16 
       48:  ret i8 %_0 
       49: } 
       50:  
       51: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(read, inaccessiblemem: readwrite) 
       52: define dso_local noundef i8 @infallible_max_unrolled(ptr noalias noundef readonly align 1 captures(none) dereferenceable(10) %x) unnamed_addr #1 { 
       53: start: 
       54:  %_23 = getelementptr inbounds nuw i8, ptr %x, i32 1 
       55:  tail call void @llvm.experimental.noalias.scope.decl(metadata !29) 
       56:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x, align 1, !alias.scope !32, !noalias !37 
       57:  tail call void @llvm.experimental.noalias.scope.decl(metadata !40) 
       58:  tail call void @llvm.experimental.noalias.scope.decl(metadata !41) 
       59:  tail call void @llvm.experimental.noalias.scope.decl(metadata !42) 
       60:  tail call void @llvm.experimental.noalias.scope.decl(metadata !43) 
       61:  %_4.val.i.i.i.i.i.i = load i8, ptr %_23, align 1, !alias.scope !37, !noalias !32, !noundef !16 
       62:  %_4.i.i.i = icmp ugt i8 %_3.val.i.i.i.i.i.pre.i, %_4.val.i.i.i.i.i.i 
       63:  %0 = tail call i8 @llvm.umax.i8(i8 %_3.val.i.i.i.i.i.pre.i, i8 %_4.val.i.i.i.i.i.i) 
       64:  %..i.i.i = select i1 %_4.i.i.i, ptr %x, ptr %_23 
       65:  %_31.i.1 = getelementptr inbounds nuw i8, ptr %x, i32 2 
       66:  %_4.val.i.i.i.i.i.i.1 = load i8, ptr %_31.i.1, align 1, !alias.scope !44, !noalias !47, !noundef !16 
       67:  %_4.i.i.i.1 = icmp ugt i8 %0, %_4.val.i.i.i.i.i.i.1 
       68:  %1 = tail call i8 @llvm.umax.i8(i8 %0, i8 %_4.val.i.i.i.i.i.i.1) 
       69:  %..i.i.i.1 = select i1 %_4.i.i.i.1, ptr %..i.i.i, ptr %_31.i.1 
       70:  %_31.i.2 = getelementptr inbounds nuw i8, ptr %x, i32 3 
       71:  %_4.val.i.i.i.i.i.i.2 = load i8, ptr %_31.i.2, align 1, !alias.scope !50, !noalias !53, !noundef !16 
       72:  %_4.i.i.i.2 = icmp ugt i8 %1, %_4.val.i.i.i.i.i.i.2 
       73:  %2 = tail call i8 @llvm.umax.i8(i8 %1, i8 %_4.val.i.i.i.i.i.i.2) 
       74:  %..i.i.i.2 = select i1 %_4.i.i.i.2, ptr %..i.i.i.1, ptr %_31.i.2 
       75:  %_31.i.3 = getelementptr inbounds nuw i8, ptr %x, i32 4 
       76:  %_4.val.i.i.i.i.i.i.3 = load i8, ptr %_31.i.3, align 1, !alias.scope !56, !noalias !59, !noundef !16 
       77:  %_4.i.i.i.3 = icmp ugt i8 %2, %_4.val.i.i.i.i.i.i.3 
       78:  %3 = tail call i8 @llvm.umax.i8(i8 %2, i8 %_4.val.i.i.i.i.i.i.3) 
       79:  %..i.i.i.3 = select i1 %_4.i.i.i.3, ptr %..i.i.i.2, ptr %_31.i.3 
       80:  %_31.i.4 = getelementptr inbounds nuw i8, ptr %x, i32 5 
       81:  %_4.val.i.i.i.i.i.i.4 = load i8, ptr %_31.i.4, align 1, !alias.scope !62, !noalias !65, !noundef !16 
       82:  %_4.i.i.i.4 = icmp ugt i8 %3, %_4.val.i.i.i.i.i.i.4 
       83:  %4 = tail call i8 @llvm.umax.i8(i8 %3, i8 %_4.val.i.i.i.i.i.i.4) 
       84:  %..i.i.i.4 = select i1 %_4.i.i.i.4, ptr %..i.i.i.3, ptr %_31.i.4 
       85:  %_31.i.5 = getelementptr inbounds nuw i8, ptr %x, i32 6 
       86:  %_4.val.i.i.i.i.i.i.5 = load i8, ptr %_31.i.5, align 1, !alias.scope !68, !noalias !71, !noundef !16 
       87:  %_4.i.i.i.5 = icmp ugt i8 %4, %_4.val.i.i.i.i.i.i.5 
       88:  %5 = tail call i8 @llvm.umax.i8(i8 %4, i8 %_4.val.i.i.i.i.i.i.5) 
       89:  %..i.i.i.5 = select i1 %_4.i.i.i.5, ptr %..i.i.i.4, ptr %_31.i.5 
       90:  %_31.i.6 = getelementptr inbounds nuw i8, ptr %x, i32 7 
       91:  %_4.val.i.i.i.i.i.i.6 = load i8, ptr %_31.i.6, align 1, !alias.scope !74, !noalias !77, !noundef !16 
       92:  %_4.i.i.i.6 = icmp ugt i8 %5, %_4.val.i.i.i.i.i.i.6 
       93:  %6 = tail call i8 @llvm.umax.i8(i8 %5, i8 %_4.val.i.i.i.i.i.i.6) 
       94:  %..i.i.i.6 = select i1 %_4.i.i.i.6, ptr %..i.i.i.5, ptr %_31.i.6 
       95:  %_31.i.7 = getelementptr inbounds nuw i8, ptr %x, i32 8 
       96:  %_4.val.i.i.i.i.i.i.7 = load i8, ptr %_31.i.7, align 1, !alias.scope !80, !noalias !83, !noundef !16 
       97:  %_4.i.i.i.7 = icmp ugt i8 %6, %_4.val.i.i.i.i.i.i.7 
       98:  %7 = tail call i8 @llvm.umax.i8(i8 %6, i8 %_4.val.i.i.i.i.i.i.7) 
       99:  %..i.i.i.7 = select i1 %_4.i.i.i.7, ptr %..i.i.i.6, ptr %_31.i.7 
      100:  %_31.i.8 = getelementptr inbounds nuw i8, ptr %x, i32 9 
      101:  %_4.val.i.i.i.i.i.i.8 = load i8, ptr %_31.i.8, align 1, !alias.scope !86, !noalias !89, !noundef !16 
      102:  %_4.i.i.i.8 = icmp ugt i8 %7, %_4.val.i.i.i.i.i.i.8 
      103:  %..i.i.i.8 = select i1 %_4.i.i.i.8, ptr %..i.i.i.7, ptr %_31.i.8 
      104:  %_0 = load i8, ptr %..i.i.i.8, align 1, !noundef !16 
      105:  ret i8 %_0 
      106: } 
      107:  
      108: ; Function Attrs: nounwind 
      109: define dso_local noundef i8 @may_panic_max(ptr noalias noundef nonnull readonly align 1 captures(none) %x.0, i32 noundef range(i32 0, -2147483648) %x.1) unnamed_addr #2 { 
      110: start: 
      111:  %_17 = icmp samesign eq i32 %x.1, 0 
      112:  br i1 %_17, label %bb2, label %bb3, !prof !92 
      113:  
      114: bb3: ; preds = %start 
      115:  %_23 = getelementptr inbounds nuw i8, ptr %x.0, i32 1 
      116:  tail call void @llvm.experimental.noalias.scope.decl(metadata !93) 
      117:  %0 = icmp samesign eq i32 %x.1, 1 
      118:  br i1 %0, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb5.i 
      119:  
      120: bb5.i: ; preds = %bb3 
      121:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x.0, align 1, !alias.scope !96, !noalias !101 
      122:  %1 = add nsw i32 %x.1, -2 
      123:  %2 = add nsw i32 %x.1, -1 
      124:  %xtraiter = and i32 %2, 3 
      125:  %3 = icmp ult i32 %1, 3 
      126:  br i1 %3, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit.loopexit.unr-lcssa, label %bb5.i.new 
      127:  
      128: bb5.i.new: ; preds = %bb5.i 
      129:  %unroll_iter = and i32 %2, -4 
      130:  %invariant.gep = getelementptr i8, ptr %_23, i32 1 
      131:  %invariant.gep2 = getelementptr i8, ptr %_23, i32 2 
      132:  %invariant.gep4 = getelementptr i8, ptr %_23, i32 3 
      133:  br label %bb10.i 
      134:  
      135: bb10.i: ; preds = %bb10.i, %bb5.i.new 
      136:  %_3.val.i.i.i.i.i.i = phi i8 [ %_3.val.i.i.i.i.i.pre.i, %bb5.i.new ], [ %7, %bb10.i ] 
      137:  %i.sroa.0.0.i = phi i32 [ 0, %bb5.i.new ], [ %_24.i.3, %bb10.i ] 
      138:  %init.sroa.0.0.i = phi ptr [ %x.0, %bb5.i.new ], [ %..i.i.i.3, %bb10.i ] 
      139:  %niter = phi i32 [ 0, %bb5.i.new ], [ %niter.next.3, %bb10.i ] 
      140:  %_31.i = getelementptr inbounds nuw i8, ptr %_23, i32 %i.sroa.0.0.i 
      141:  tail call void @llvm.experimental.noalias.scope.decl(metadata !104) 
      142:  tail call void @llvm.experimental.noalias.scope.decl(metadata !105) 
      143:  tail call void @llvm.experimental.noalias.scope.decl(metadata !106) 
      144:  tail call void @llvm.experimental.noalias.scope.decl(metadata !107) 
        .
        .
        .
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/issues/iter-max-no-unwrap-failed-129583/iter-max-no-unwrap-failed-129583.ll" "/checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs:13:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: unwrap_failed
               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/issues/iter-max-no-unwrap-failed-129583/iter-max-no-unwrap-failed-129583.ll:44:213: note: found here
 br i1 %_25.i.2, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb10.i
                                                                                                                                                                                                                    ^~~~~~~~~~~~~

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/issues/iter-max-no-unwrap-failed-129583/iter-max-no-unwrap-failed-129583.ll
Check file: /checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
        1: ; ModuleID = 'iter_max_no_unwrap_failed_129583.5c466b832002e549-cgu.0' 
        2: source_filename = "iter_max_no_unwrap_failed_129583.5c466b832002e549-cgu.0" 
        3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-i128:128-n32:64-S128-ni:1:10:20" 
        4: target triple = "wasm32-unknown-wasip1" 
        5:  
        6: @alloc_be0d87cab5c3b8ce747aaa7e9520f67f = private unnamed_addr constant [72 x i8] c"/checkout/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs\00", align 1 
        7: @alloc_ec844ed9573175be2fa8ca9555f08e2e = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_be0d87cab5c3b8ce747aaa7e9520f67f, [12 x i8] c"G\00\00\00\1D\00\00\00\15\00\00\00" }>, align 4 
        8:  
        9: ; Function Attrs: nofree norecurse nosync nounwind memory(argmem: read, inaccessiblemem: readwrite) 
       10: define dso_local noundef i8 @infallible_max_not_unrolled(ptr noalias noundef readonly align 1 captures(none) dereferenceable(1024) %x) unnamed_addr #0 { 
       11: start: 
       12:  %_23 = getelementptr inbounds nuw i8, ptr %x, i32 1 
       13:  tail call void @llvm.experimental.noalias.scope.decl(metadata !1) 
       14:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x, align 1, !alias.scope !4, !noalias !9 
       15:  %invariant.gep = getelementptr inbounds i8, ptr %_23, i32 1 
       16:  %invariant.gep1 = getelementptr inbounds i8, ptr %_23, i32 2 
       17:  br label %bb10.i 
       18:  
       19: bb10.i: ; preds = %bb10.i, %start 
       20:  %_3.val.i.i.i.i.i.i = phi i8 [ %_3.val.i.i.i.i.i.pre.i, %start ], [ %2, %bb10.i ] 
       21:  %i.sroa.0.0.i = phi i32 [ 0, %start ], [ %_24.i.2, %bb10.i ] 
       22:  %init.sroa.0.0.i = phi ptr [ %x, %start ], [ %..i.i.i.2, %bb10.i ] 
       23:  %_31.i = getelementptr inbounds nuw i8, ptr %_23, i32 %i.sroa.0.0.i 
       24:  tail call void @llvm.experimental.noalias.scope.decl(metadata !12) 
       25:  tail call void @llvm.experimental.noalias.scope.decl(metadata !13) 
       26:  tail call void @llvm.experimental.noalias.scope.decl(metadata !14) 
       27:  tail call void @llvm.experimental.noalias.scope.decl(metadata !15) 
       28:  %_4.val.i.i.i.i.i.i = load i8, ptr %_31.i, align 1, !alias.scope !9, !noalias !4, !noundef !16 
       29:  %_4.i.i.i = icmp ugt i8 %_3.val.i.i.i.i.i.i, %_4.val.i.i.i.i.i.i 
       30:  %0 = tail call i8 @llvm.umax.i8(i8 %_3.val.i.i.i.i.i.i, i8 %_4.val.i.i.i.i.i.i) 
       31:  %..i.i.i = select i1 %_4.i.i.i, ptr %init.sroa.0.0.i, ptr %_31.i 
       32:  %gep = getelementptr inbounds i8, ptr %invariant.gep, i32 %i.sroa.0.0.i 
       33:  %_4.val.i.i.i.i.i.i.1 = load i8, ptr %gep, align 1, !alias.scope !17, !noalias !20, !noundef !16 
       34:  %_4.i.i.i.1 = icmp ugt i8 %0, %_4.val.i.i.i.i.i.i.1 
       35:  %1 = tail call i8 @llvm.umax.i8(i8 %0, i8 %_4.val.i.i.i.i.i.i.1) 
       36:  %..i.i.i.1 = select i1 %_4.i.i.i.1, ptr %..i.i.i, ptr %gep 
       37:  %gep2 = getelementptr inbounds i8, ptr %invariant.gep1, i32 %i.sroa.0.0.i 
       38:  %_4.val.i.i.i.i.i.i.2 = load i8, ptr %gep2, align 1, !alias.scope !23, !noalias !26, !noundef !16 
       39:  %_4.i.i.i.2 = icmp ugt i8 %1, %_4.val.i.i.i.i.i.i.2 
       40:  %2 = tail call i8 @llvm.umax.i8(i8 %1, i8 %_4.val.i.i.i.i.i.i.2) 
       41:  %..i.i.i.2 = select i1 %_4.i.i.i.2, ptr %..i.i.i.1, ptr %gep2 
       42:  %_24.i.2 = add nuw nsw i32 %i.sroa.0.0.i, 3 
       43:  %_25.i.2 = icmp eq i32 %_24.i.2, 1023 
       44:  br i1 %_25.i.2, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb10.i 
not:13                                                                                                                                                                                                                         !~~~~~~~~~~~~                             error: no match expected
       45:  
       46: _RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit: ; preds = %bb10.i 
       47:  %_0 = load i8, ptr %..i.i.i.2, align 1, !noundef !16 
       48:  ret i8 %_0 
       49: } 
       50:  
       51: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(read, inaccessiblemem: readwrite) 
       52: define dso_local noundef i8 @infallible_max_unrolled(ptr noalias noundef readonly align 1 captures(none) dereferenceable(10) %x) unnamed_addr #1 { 
       53: start: 
       54:  %_23 = getelementptr inbounds nuw i8, ptr %x, i32 1 
       55:  tail call void @llvm.experimental.noalias.scope.decl(metadata !29) 
       56:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x, align 1, !alias.scope !32, !noalias !37 
       57:  tail call void @llvm.experimental.noalias.scope.decl(metadata !40) 
       58:  tail call void @llvm.experimental.noalias.scope.decl(metadata !41) 
       59:  tail call void @llvm.experimental.noalias.scope.decl(metadata !42) 
       60:  tail call void @llvm.experimental.noalias.scope.decl(metadata !43) 
       61:  %_4.val.i.i.i.i.i.i = load i8, ptr %_23, align 1, !alias.scope !37, !noalias !32, !noundef !16 
       62:  %_4.i.i.i = icmp ugt i8 %_3.val.i.i.i.i.i.pre.i, %_4.val.i.i.i.i.i.i 
       63:  %0 = tail call i8 @llvm.umax.i8(i8 %_3.val.i.i.i.i.i.pre.i, i8 %_4.val.i.i.i.i.i.i) 
       64:  %..i.i.i = select i1 %_4.i.i.i, ptr %x, ptr %_23 
       65:  %_31.i.1 = getelementptr inbounds nuw i8, ptr %x, i32 2 
       66:  %_4.val.i.i.i.i.i.i.1 = load i8, ptr %_31.i.1, align 1, !alias.scope !44, !noalias !47, !noundef !16 
       67:  %_4.i.i.i.1 = icmp ugt i8 %0, %_4.val.i.i.i.i.i.i.1 
       68:  %1 = tail call i8 @llvm.umax.i8(i8 %0, i8 %_4.val.i.i.i.i.i.i.1) 
       69:  %..i.i.i.1 = select i1 %_4.i.i.i.1, ptr %..i.i.i, ptr %_31.i.1 
       70:  %_31.i.2 = getelementptr inbounds nuw i8, ptr %x, i32 3 
       71:  %_4.val.i.i.i.i.i.i.2 = load i8, ptr %_31.i.2, align 1, !alias.scope !50, !noalias !53, !noundef !16 
       72:  %_4.i.i.i.2 = icmp ugt i8 %1, %_4.val.i.i.i.i.i.i.2 
       73:  %2 = tail call i8 @llvm.umax.i8(i8 %1, i8 %_4.val.i.i.i.i.i.i.2) 
       74:  %..i.i.i.2 = select i1 %_4.i.i.i.2, ptr %..i.i.i.1, ptr %_31.i.2 
       75:  %_31.i.3 = getelementptr inbounds nuw i8, ptr %x, i32 4 
       76:  %_4.val.i.i.i.i.i.i.3 = load i8, ptr %_31.i.3, align 1, !alias.scope !56, !noalias !59, !noundef !16 
       77:  %_4.i.i.i.3 = icmp ugt i8 %2, %_4.val.i.i.i.i.i.i.3 
       78:  %3 = tail call i8 @llvm.umax.i8(i8 %2, i8 %_4.val.i.i.i.i.i.i.3) 
       79:  %..i.i.i.3 = select i1 %_4.i.i.i.3, ptr %..i.i.i.2, ptr %_31.i.3 
       80:  %_31.i.4 = getelementptr inbounds nuw i8, ptr %x, i32 5 
       81:  %_4.val.i.i.i.i.i.i.4 = load i8, ptr %_31.i.4, align 1, !alias.scope !62, !noalias !65, !noundef !16 
       82:  %_4.i.i.i.4 = icmp ugt i8 %3, %_4.val.i.i.i.i.i.i.4 
       83:  %4 = tail call i8 @llvm.umax.i8(i8 %3, i8 %_4.val.i.i.i.i.i.i.4) 
       84:  %..i.i.i.4 = select i1 %_4.i.i.i.4, ptr %..i.i.i.3, ptr %_31.i.4 
       85:  %_31.i.5 = getelementptr inbounds nuw i8, ptr %x, i32 6 
       86:  %_4.val.i.i.i.i.i.i.5 = load i8, ptr %_31.i.5, align 1, !alias.scope !68, !noalias !71, !noundef !16 
       87:  %_4.i.i.i.5 = icmp ugt i8 %4, %_4.val.i.i.i.i.i.i.5 
       88:  %5 = tail call i8 @llvm.umax.i8(i8 %4, i8 %_4.val.i.i.i.i.i.i.5) 
       89:  %..i.i.i.5 = select i1 %_4.i.i.i.5, ptr %..i.i.i.4, ptr %_31.i.5 
       90:  %_31.i.6 = getelementptr inbounds nuw i8, ptr %x, i32 7 
       91:  %_4.val.i.i.i.i.i.i.6 = load i8, ptr %_31.i.6, align 1, !alias.scope !74, !noalias !77, !noundef !16 
       92:  %_4.i.i.i.6 = icmp ugt i8 %5, %_4.val.i.i.i.i.i.i.6 
       93:  %6 = tail call i8 @llvm.umax.i8(i8 %5, i8 %_4.val.i.i.i.i.i.i.6) 
       94:  %..i.i.i.6 = select i1 %_4.i.i.i.6, ptr %..i.i.i.5, ptr %_31.i.6 
       95:  %_31.i.7 = getelementptr inbounds nuw i8, ptr %x, i32 8 
       96:  %_4.val.i.i.i.i.i.i.7 = load i8, ptr %_31.i.7, align 1, !alias.scope !80, !noalias !83, !noundef !16 
       97:  %_4.i.i.i.7 = icmp ugt i8 %6, %_4.val.i.i.i.i.i.i.7 
       98:  %7 = tail call i8 @llvm.umax.i8(i8 %6, i8 %_4.val.i.i.i.i.i.i.7) 
       99:  %..i.i.i.7 = select i1 %_4.i.i.i.7, ptr %..i.i.i.6, ptr %_31.i.7 
      100:  %_31.i.8 = getelementptr inbounds nuw i8, ptr %x, i32 9 
      101:  %_4.val.i.i.i.i.i.i.8 = load i8, ptr %_31.i.8, align 1, !alias.scope !86, !noalias !89, !noundef !16 
      102:  %_4.i.i.i.8 = icmp ugt i8 %7, %_4.val.i.i.i.i.i.i.8 
      103:  %..i.i.i.8 = select i1 %_4.i.i.i.8, ptr %..i.i.i.7, ptr %_31.i.8 
      104:  %_0 = load i8, ptr %..i.i.i.8, align 1, !noundef !16 
      105:  ret i8 %_0 
      106: } 
      107:  
      108: ; Function Attrs: nounwind 
      109: define dso_local noundef i8 @may_panic_max(ptr noalias noundef nonnull readonly align 1 captures(none) %x.0, i32 noundef range(i32 0, -2147483648) %x.1) unnamed_addr #2 { 
      110: start: 
      111:  %_17 = icmp samesign eq i32 %x.1, 0 
      112:  br i1 %_17, label %bb2, label %bb3, !prof !92 
      113:  
      114: bb3: ; preds = %start 
      115:  %_23 = getelementptr inbounds nuw i8, ptr %x.0, i32 1 
      116:  tail call void @llvm.experimental.noalias.scope.decl(metadata !93) 
      117:  %0 = icmp samesign eq i32 %x.1, 1 
      118:  br i1 %0, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit, label %bb5.i 
      119:  
      120: bb5.i: ; preds = %bb3 
      121:  %_3.val.i.i.i.i.i.pre.i = load i8, ptr %x.0, align 1, !alias.scope !96, !noalias !101 
      122:  %1 = add nsw i32 %x.1, -2 
      123:  %2 = add nsw i32 %x.1, -1 
      124:  %xtraiter = and i32 %2, 3 
      125:  %3 = icmp ult i32 %1, 3 
      126:  br i1 %3, label %_RINvXs2E_NtNtCs1i4ptGVylu2_4core5slice4iterINtB7_4IterhENtNtNtNtBb_4iter6traits8iterator8Iterator4foldRhNCINvNvBS_6max_by4foldB1C_NvYB1C_NtNtBb_3cmp3Ord3cmpE0ECs7Vb2wfTRDQH_32iter_max_no_unwrap_failed_129583.exit.loopexit.unr-lcssa, label %bb5.i.new 
      127:  
      128: bb5.i.new: ; preds = %bb5.i 
      129:  %unroll_iter = and i32 %2, -4 
      130:  %invariant.gep = getelementptr i8, ptr %_23, i32 1 
      131:  %invariant.gep2 = getelementptr i8, ptr %_23, i32 2 
      132:  %invariant.gep4 = getelementptr i8, ptr %_23, i32 3 
      133:  br label %bb10.i 
      134:  
      135: bb10.i: ; preds = %bb10.i, %bb5.i.new 
      136:  %_3.val.i.i.i.i.i.i = phi i8 [ %_3.val.i.i.i.i.i.pre.i, %bb5.i.new ], [ %7, %bb10.i ] 
      137:  %i.sroa.0.0.i = phi i32 [ 0, %bb5.i.new ], [ %_24.i.3, %bb10.i ] 
      138:  %init.sroa.0.0.i = phi ptr [ %x.0, %bb5.i.new ], [ %..i.i.i.3, %bb10.i ] 
      139:  %niter = phi i32 [ 0, %bb5.i.new ], [ %niter.next.3, %bb10.i ] 
      140:  %_31.i = getelementptr inbounds nuw i8, ptr %_23, i32 %i.sroa.0.0.i 
      141:  tail call void @llvm.experimental.noalias.scope.decl(metadata !104) 
      142:  tail call void @llvm.experimental.noalias.scope.decl(metadata !105) 
      143:  tail call void @llvm.experimental.noalias.scope.decl(metadata !106) 
      144:  tail call void @llvm.experimental.noalias.scope.decl(metadata !107) 
        .
        .
        .
>>>>>>
------------------------------------------
```

</details>

--- a/compiler/rustc_session/src/config.rs
+++ b/compiler/rustc_session/src/config.rs
@@ -1530,7 +1530,11 @@ impl Options {
     }
 
     pub fn get_symbol_mangling_version(&self) -> SymbolManglingVersion {
-        self.cg.symbol_mangling_version.unwrap_or(SymbolManglingVersion::Legacy)
+        self.cg.symbol_mangling_version.unwrap_or(if self.unstable_features.is_nightly_build() {
+            SymbolManglingVersion::V0
+        } else {
+            SymbolManglingVersion::Legacy
+        })
     }
 
     #[inline]

--- a/src/doc/rustc/src/symbol-mangling/index.md
+++ b/src/doc/rustc/src/symbol-mangling/index.md
@@ -47,6 +47,6 @@ foo::example_function
 ## Mangling versions
 
 `rustc` supports different mangling versions which encode the names in different ways.
-The legacy version (which is currently the default) is not described here.
+The legacy version (which is currently the default on beta/stable) is not described here.
 The "v0" mangling scheme addresses several limitations of the legacy format,
 and is described in the [v0 Symbol Format](v0.md) chapter.

--- a/tests/assembly-llvm/closure-inherit-target-feature.rs
+++ b/tests/assembly-llvm/closure-inherit-target-feature.rs
@@ -13,7 +13,7 @@ use std::arch::x86_64::{__m128, _mm_blend_ps};
 pub unsafe fn sse41_blend_nofeature(x: __m128, y: __m128, ret: *mut __m128) {
     let f = {
         // check that _mm_blend_ps is not being inlined into the closure
-        // CHECK-LABEL: {{sse41_blend_nofeature.*closure.*:}}
+        // CHECK-LABEL: {{sse41_blend_nofeature:}}
         // CHECK-NOT: blendps
         // CHECK: {{call .*_mm_blend_ps.*}}
         // CHECK-NOT: blendps
@@ -29,7 +29,7 @@ pub unsafe fn sse41_blend_nofeature(x: __m128, y: __m128, ret: *mut __m128) {
 pub fn sse41_blend_noinline(x: __m128, y: __m128) -> __m128 {
     let f = {
         // check that _mm_blend_ps is being inlined into the closure
-        // CHECK-LABEL: {{sse41_blend_noinline.*closure.*:}}
+        // CHECK-LABEL: {{sse41_blend_noinline:}}
         // CHECK-NOT: _mm_blend_ps
         // CHECK: blendps
         // CHECK-NOT: _mm_blend_ps
@@ -45,10 +45,10 @@ pub fn sse41_blend_noinline(x: __m128, y: __m128) -> __m128 {
 pub fn sse41_blend_doinline(x: __m128, y: __m128) -> __m128 {
     // check that the closure and _mm_blend_ps are being inlined into the function
     // CHECK-LABEL: sse41_blend_doinline:
-    // CHECK-NOT: {{sse41_blend_doinline.*closure.*}}
+    // CHECK-NOT: {{sse41_blend_doinline:}}
     // CHECK-NOT: _mm_blend_ps
     // CHECK: blendps
-    // CHECK-NOT: {{sse41_blend_doinline.*closure.*}}
+    // CHECK-NOT: {{sse41_blend_doinline.*}}
     // CHECK-NOT: _mm_blend_ps
     // CHECK: ret
     let f = {

--- a/tests/assembly-llvm/nvptx-safe-naming.rs
+++ b/tests/assembly-llvm/nvptx-safe-naming.rs
@@ -12,7 +12,7 @@
 extern crate breakpoint_panic_handler;
 
 // Verify function name doesn't contain unacceaptable characters.
-// CHECK: .func (.param .b32 func_retval0) [[IMPL_FN:[a-zA-Z0-9$_]+square[a-zA-Z0-9$_]+]]
+// CHECK: .func (.param .b32 func_retval0) [[IMPL_FN:[a-zA-Z0-9$_]+square]]
 
 // CHECK-LABEL: .visible .entry top_kernel(
 #[no_mangle]

--- a/tests/codegen-llvm/array-from_fn.rs
+++ b/tests/codegen-llvm/array-from_fn.rs
@@ -7,7 +7,7 @@
 
 #[no_mangle]
 pub fn iota() -> [u8; 16] {
-    // OPT-NOT: core..array..Guard
-    // NORMAL: core..array..Guard
+    // OPT-NOT: core::array::Guard
+    // NORMAL: core::array::Guard
     std::array::from_fn(|i| i as _)
 }

--- a/tests/codegen-llvm/binary-heap-peek-mut-pop-no-panic.rs
+++ b/tests/codegen-llvm/binary-heap-peek-mut-pop-no-panic.rs
@@ -1,4 +1,4 @@
-//@ compile-flags: -Copt-level=3
+//@ compile-flags: -Copt-level=3 --crate-name=test
 //@ ignore-std-debug-assertions
 #![crate_type = "lib"]
 

--- a/tests/codegen-llvm/cold-attribute.rs
+++ b/tests/codegen-llvm/cold-attribute.rs
@@ -20,7 +20,7 @@ pub async fn async_block() {
         f.await;
     }
     x(
-        // CHECK-LABEL: ; cold_attribute::async_block::{{{{closure}}}}::{{{{closure}}}}
+        // CHECK-LABEL: ; cold_attribute::async_block::{closure#0}::{closure#0}
         // CHECK-NEXT: Function Attrs: cold {{.*}}
         #[cold]
         async {},
@@ -33,7 +33,7 @@ pub fn closure() {
         f()
     }
     x(
-        // CHECK-LABEL: ; cold_attribute::closure::{{{{closure}}}}
+        // CHECK-LABEL: ; cold_attribute::closure::{closure#0}
         // CHECK-NEXT: Function Attrs: cold {{.*}}
         #[cold]
         || {},
@@ -43,14 +43,14 @@ pub fn closure() {
 pub struct S;
 
 impl S {
-    // CHECK-LABEL: ; cold_attribute::S::method
+    // CHECK-LABEL: ; <cold_attribute::S>::method
     // CHECK-NEXT: Function Attrs: cold {{.*}}
     #[cold]
     pub fn method(&self) {}
 }
 
 pub trait Trait {
-    // CHECK-LABEL: ; cold_attribute::Trait::trait_fn
+    // CHECK-LABEL: ; <cold_attribute::S as cold_attribute::Trait>::trait_fn
     // CHECK-NEXT: Function Attrs: cold {{.*}}
     #[cold]
     fn trait_fn(&self) {}

--- a/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs
+++ b/tests/codegen-llvm/issues/iter-max-no-unwrap-failed-129583.rs
@@ -2,7 +2,7 @@
 // The iterator may unroll for values smaller than a certain threshold so we
 // use a larger value to prevent unrolling.
 
-//@ compile-flags: -Copt-level=3
+//@ compile-flags: -Copt-level=3 --crate-name=test
 
 #![crate_type = "lib"]
 

--- a/tests/codegen-llvm/naked-fn/generics.rs
+++ b/tests/codegen-llvm/naked-fn/generics.rs
@@ -20,11 +20,11 @@ fn test(x: u64) {
 }
 
 // CHECK: .balign 4
-// CHECK: add rax, 2
+// CHECK: add rax, 1
 // CHECK: add rax, 42
 
 // CHECK: .balign 4
-// CHECK: add rax, 1
+// CHECK: add rax, 2
 // CHECK: add rax, 42
 
 #[unsafe(naked)]

--- a/tests/codegen-llvm/no-alloca-inside-if-false.rs
+++ b/tests/codegen-llvm/no-alloca-inside-if-false.rs
@@ -7,10 +7,10 @@
 
 #[inline(never)]
 fn test<const SIZE: usize>() {
-    // CHECK-LABEL: no_alloca_inside_if_false::test
+    // CHECK-LABEL: no_alloca_inside_if_false::test::<8192>
     // CHECK: start:
-    // CHECK-NEXT: alloca [{{12|24}} x i8]
-    // CHECK-NOT: alloca
+    // CHECK-NEXT: = alloca [{{12|24}} x i8]
+    // CHECK-NOT: = alloca
     if const { SIZE < 4096 } {
         let arr = [0u8; SIZE];
         std::hint::black_box(&arr);

--- a/tests/codegen-llvm/precondition-checks.rs
+++ b/tests/codegen-llvm/precondition-checks.rs
@@ -13,13 +13,13 @@
 
 use std::ptr::NonNull;
 
-// CHECK-LABEL: ; core::ptr::non_null::NonNull<T>::new_unchecked
+// CHECK-LABEL: ; <core::ptr::non_null::NonNull<u8>>::new_unchecked
 // CHECK-NOT: call
 // CHECK: }
 
 // CHECK-LABEL: @nonnull_new
 #[no_mangle]
 pub unsafe fn nonnull_new(ptr: *mut u8) -> NonNull<u8> {
-    // CHECK: ; call core::ptr::non_null::NonNull<T>::new_unchecked
+    // CHECK: ; call <core::ptr::non_null::NonNull<u8>>::new_unchecked
     unsafe { NonNull::new_unchecked(ptr) }
 }

--- a/tests/codegen-llvm/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-drop-in-place.rs
+++ b/tests/codegen-llvm/sanitizer/cfi/emit-type-metadata-id-itanium-cxx-abi-drop-in-place.rs
@@ -9,7 +9,7 @@
 
 #![crate_type = "lib"]
 
-// CHECK-LABEL: define{{.*}}4core3ptr47drop_in_place$LT$dyn$u20$core..marker..Send$GT$
+// CHECK-LABEL: define{{.*}}4core3ptr13drop_in_placeDNtNtB4_6marker4Send
 // CHECK-SAME:  {{.*}}!type ![[TYPE1:[0-9]+]] !type !{{[0-9]+}} !type !{{[0-9]+}} !type !{{[0-9]+}}
 // CHECK:       call i1 @llvm.type.test(ptr {{%.+}}, metadata !"_ZTSFvPu3dynIu{{[0-9]+}}NtNtNtC{{[[:print:]]+}}_4core3ops4drop4Dropu6regionEE")
 
@@ -20,7 +20,7 @@ struct PresentDrop;
 
 impl Drop for PresentDrop {
     fn drop(&mut self) {}
-    // CHECK: define{{.*}}4core3ptr{{[0-9]+}}drop_in_place$LT${{.*}}PresentDrop$GT${{.*}}!type ![[TYPE1]] !type !{{[0-9]+}} !type !{{[0-9]+}} !type !{{[0-9]+}}
+    // CHECK: define{{.*}}4core3ptr{{[0-9]+}}drop_in_place{{.*}}PresentDrop{{.*}}!type ![[TYPE1]] !type !{{[0-9]+}} !type !{{[0-9]+}} !type !{{[0-9]+}}
 }
 
 pub fn foo() {

--- a/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs
+++ b/tests/codegen-llvm/sanitizer/kcfi/fn-ptr-reify-shim.rs
@@ -54,11 +54,11 @@ pub fn main() {
     // `DynCompatible` is indeed dyn-compatible.
     let _: &dyn DynCompatible = &s;
 
-    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name{{.*}}reify.shim.fnptr
+    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name::{shim:reify_fnptr#0}
     let dyn_name = S::dyn_name as fn(&S) -> &str;
     let _unused = dyn_name(&s);
 
-    // CHECK: call fn_ptr_reify_shim::DynCompatible::dyn_name_default{{.*}}reify.shim.fnptr
+    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::DynCompatible>::dyn_name_default::{shim:reify_fnptr#0}
     let dyn_name_default = S::dyn_name_default as fn(&S) -> &str;
     let _unused = dyn_name_default(&s);
 
@@ -68,7 +68,7 @@ pub fn main() {
     let not_dyn_name = S::not_dyn_name as fn() -> &'static str;
     let _unused = not_dyn_name();
 
-    // CHECK: call fn_ptr_reify_shim::NotDynCompatible::not_dyn_name_default{{$}}
+    // CHECK: call <fn_ptr_reify_shim::S as fn_ptr_reify_shim::NotDynCompatible>::not_dyn_name_default{{$}}
     let not_dyn_name_default = S::not_dyn_name_default as fn() -> &'static str;
     let _unused = not_dyn_name_default();
 }

--- a/tests/codegen-llvm/sanitizer/kcfi/naked-function.rs
+++ b/tests/codegen-llvm/sanitizer/kcfi/naked-function.rs
@@ -29,7 +29,7 @@ impl MyTrait for Thing {}
 // the shim calls the real function
 // CHECK-LABEL: define
 // CHECK-SAME: my_naked_function
-// CHECK-SAME: reify.shim.fnptr
+// CHECK-SAME: reify_fnptr
 
 // CHECK-LABEL: main
 #[unsafe(no_mangle)]
@@ -40,7 +40,7 @@ pub fn main() {
     // main calls the shim function
     // CHECK: call void
     // CHECK-SAME: my_naked_function
-    // CHECK-SAME: reify.shim.fnptr
+    // CHECK-SAME: reify_fnptr
     (F)(&Thing);
 }
 

--- a/tests/codegen-llvm/sanitizer/sanitize-off.rs
+++ b/tests/codegen-llvm/sanitizer/sanitize-off.rs
@@ -66,7 +66,7 @@ pub trait MyTrait {
     fn unsanitized(&self, b: &mut u8) -> u8;
     fn sanitized(&self, b: &mut u8) -> u8;
 
-    // CHECK-LABEL: ; sanitize_off::MyTrait::unsanitized_default
+    // CHECK-LABEL: ; <() as sanitize_off::MyTrait>::unsanitized_default
     // CHECK-NEXT:  ; Function Attrs:
     // CHECK-NOT:   sanitize_address
     // CHECK:       start:
@@ -77,7 +77,7 @@ pub trait MyTrait {
         *b
     }
 
-    // CHECK-LABEL: ; sanitize_off::MyTrait::sanitized_default
+    // CHECK-LABEL: ; <() as sanitize_off::MyTrait>::sanitized_default
     // CHECK-NEXT:  ; Function Attrs:
     // CHECK:       sanitize_address
     // CHECK:       start:

--- a/tests/debuginfo/basic-types-globals.rs
+++ b/tests/debuginfo/basic-types-globals.rs
@@ -9,35 +9,35 @@
 
 // lldb-command:run
 // lldb-command:v B
-// lldb-check: ::B::[...] = false
+// lldb-check: ::B = false
 // lldb-command:v I
-// lldb-check: ::I::[...] = -1
+// lldb-check: ::I = -1
 // lldb-command:v --format=d C
-// lldb-check: ::C::[...] = 97
+// lldb-check: ::C = 97
 // lldb-command:v --format=d I8
-// lldb-check: ::I8::[...] = 68
+// lldb-check: ::I8 = 68
 // lldb-command:v I16
-// lldb-check: ::I16::[...] = -16
+// lldb-check: ::I16 = -16
 // lldb-command:v I32
-// lldb-check: ::I32::[...] = -32
+// lldb-check: ::I32 = -32
 // lldb-command:v I64
-// lldb-check: ::I64::[...] = -64
+// lldb-check: ::I64 = -64
 // lldb-command:v U
-// lldb-check: ::U::[...] = 1
+// lldb-check: ::U = 1
 // lldb-command:v --format=d U8
-// lldb-check: ::U8::[...] = 100
+// lldb-check: ::U8 = 100
 // lldb-command:v U16
-// lldb-check: ::U16::[...] = 16
+// lldb-check: ::U16 = 16
 // lldb-command:v U32
-// lldb-check: ::U32::[...] = 32
+// lldb-check: ::U32 = 32
 // lldb-command:v U64
-// lldb-check: ::U64::[...] = 64
+// lldb-check: ::U64 = 64
 // lldb-command:v F16
-// lldb-check: ::F16::[...] = 1.5
+// lldb-check: ::F16 = 1.5
 // lldb-command:v F32
-// lldb-check: ::F32::[...] = 2.5
+// lldb-check: ::F32 = 2.5
 // lldb-command:v F64
-// lldb-check: ::F64::[...] = 3.5
+// lldb-check: ::F64 = 3.5
 
 // gdb-command:run
 // gdb-command:print B
@@ -103,4 +103,6 @@ fn main() {
     let b = unsafe { F16 };
 }
 
-fn _zzz() {()}
+fn _zzz() {
+    ()
+}

--- a/tests/debuginfo/no_mangle-info.rs
+++ b/tests/debuginfo/no_mangle-info.rs
@@ -14,7 +14,7 @@
 // lldb-command:v TEST
 // lldb-check:(unsigned long) TEST = 3735928559
 // lldb-command:v OTHER_TEST
-// lldb-check:(unsigned long) no_mangle_info::namespace::OTHER_TEST::[...] = 42
+// lldb-check:(unsigned long) no_mangle_info::namespace::OTHER_TEST = 42
 
 // === CDB TESTS ==================================================================================
 // cdb-command: g

--- a/tests/ui/panics/short-ice-remove-middle-frames-2.run.stderr
+++ b/tests/ui/panics/short-ice-remove-middle-frames-2.run.stderr
@@ -4,12 +4,12 @@ debug!!!
 stack backtrace:
    0: std::panicking::begin_panic
    1: short_ice_remove_middle_frames_2::eight
-   2: short_ice_remove_middle_frames_2::seven::{{closure}}
+   2: short_ice_remove_middle_frames_2::seven::{closure#0}
       [... omitted 3 frames ...]
    3: short_ice_remove_middle_frames_2::fifth
-   4: short_ice_remove_middle_frames_2::fourth::{{closure}}
+   4: short_ice_remove_middle_frames_2::fourth::{closure#0}
       [... omitted 4 frames ...]
    5: short_ice_remove_middle_frames_2::first
    6: short_ice_remove_middle_frames_2::main
-   7: core::ops::function::FnOnce::call_once
+   7: <fn() as core::ops::function::FnOnce<()>>::call_once
 note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

--- a/tests/ui/panics/short-ice-remove-middle-frames.run.stderr
+++ b/tests/ui/panics/short-ice-remove-middle-frames.run.stderr
@@ -5,11 +5,11 @@ stack backtrace:
    0: std::panicking::begin_panic
    1: short_ice_remove_middle_frames::seven
    2: short_ice_remove_middle_frames::sixth
-   3: short_ice_remove_middle_frames::fifth::{{closure}}
+   3: short_ice_remove_middle_frames::fifth::{closure#0}
       [... omitted 4 frames ...]
    4: short_ice_remove_middle_frames::second
-   5: short_ice_remove_middle_frames::first::{{closure}}
+   5: short_ice_remove_middle_frames::first::{closure#0}
    6: short_ice_remove_middle_frames::first
    7: short_ice_remove_middle_frames::main
-   8: core::ops::function::FnOnce::call_once
+   8: <fn() as core::ops::function::FnOnce<()>>::call_once
 note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

--- a/tests/ui/sanitizer/realtime-blocking.rs
+++ b/tests/ui/sanitizer/realtime-blocking.rs
@@ -6,7 +6,7 @@
 //
 //@ run-fail
 //@ error-pattern: Call to blocking function
-//@ error-pattern: realtime_blocking::blocking::
+//@ error-pattern: realtime_blocking::blocking
 //@ ignore-backends: gcc
 #![feature(sanitize)]
 

